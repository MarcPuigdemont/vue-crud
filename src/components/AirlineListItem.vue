<template>
  <b-list-group-item class="flex-column align-items-start" style="cursor: pointer" data-test="list-row" @click="$emit('edit', airline)">
    <b-row>
      <b-col data-test="list-row-iata">
        {{ airline.iata }}
      </b-col>
      <b-col data-test="list-row-name">
        {{ airline.name }}
      </b-col>
      <b-col :title="airline.primary_color">
        <div :style="{ 'background-color': airline.primary_color }" class="h-100" />
      </b-col>
      <b-col :title="airline.secondary_color">
        <div :style="{ 'background-color': airline.secondary_color }" class="h-100" />
      </b-col>
      <b-col>
        <AirlineListItemServices :services="servicesList" />
      </b-col>
    </b-row>
  </b-list-group-item>
</template>

<script>
import AirlineListItemServices from './AirlineListItemServices.vue';

export default {
  name: 'AirlineListItem',
  props: {
    airline: Object,
  },
  components: {
    AirlineListItemServices,
  },
  computed: {
    servicesList: function() {
      return Object.keys(this.airline.services).map(service => ({ name: service, available: this.airline.services[service] }));
    },
  },
};
</script>
