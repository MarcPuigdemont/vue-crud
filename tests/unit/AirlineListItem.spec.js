import { mount } from '@vue/test-utils';
import AirlineListItem from '@/components/AirlineListItem.vue';

describe('AirlineListItem', () => {
  test('should match the snapshot', () => {
    const propsData = {
      airline: {
        iata: 'BX',
        name: 'Airline name',
        primary_color: '#ffffff',
        secondary_color: '#000000',
        services: { bags: true, checkin: false, seats: true },
      },
    };
    const wrapper = mount(AirlineListItem, {
      propsData,
      stubs: {
        AirlineListItemServices: '<div class="airline-list-item-services"></div>',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    const expectedServicesList = [{ name: 'bags', available: true }, { name: 'checkin', available: false }, { name: 'seats', available: true }];
    expect(wrapper.vm.servicesList).toEqual(expectedServicesList);
  });
});
