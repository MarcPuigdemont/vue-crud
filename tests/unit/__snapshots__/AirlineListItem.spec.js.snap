// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineListItem should match the snapshot 1`] = `
<div
  class="list-group-item flex-column align-items-start"
  data-test="list-row"
  style="cursor: pointer;"
>
  <div
    class="row"
  >
    <div
      class="col"
      data-test="list-row-iata"
    >
      
      BX
    
    </div>
     
    <div
      class="col"
      data-test="list-row-name"
    >
      
      Airline name
    
    </div>
     
    <div
      class="col"
      title="#ffffff"
    >
      <div
        class="h-100"
        style="background-color: rgb(255, 255, 255);"
      />
    </div>
     
    <div
      class="col"
      title="#000000"
    >
      <div
        class="h-100"
        style="background-color: rgb(0, 0, 0);"
      />
    </div>
     
    <div
      class="col"
    >
      <div
        class="airline-list-item-services"
      />
    </div>
  </div>
</div>
`;
